Toolkit/ArcGISToolkit/JobManager.swift

```diff
@@ -212,6 +217,58 @@ public class JobManager: NSObject {
         }
     }
     
+    private let bgTaskIdentifier = "com.esri.arcgis.toolkit.jobmanager.refresh"
+    private var bgObserver: AnyObject?
+    
+    /// Registers a task for updating job status in the background.
+    /// You must add an entry in the app plist for com.esri.arcgis.toolkit.jobmanager.refresh under BGTaskSchedulerPermittedIdentifiers.
+    /// This must be called before the end of the app launch sequence.
+    /// This must be tested on device, does not work on the simulator. For debugging this, see documentation here: https://developer.apple.com/documentation/backgroundtasks/starting_and_terminating_tasks_during_development?language=objc
+    @available(iOS 13.0, *)
+    @discardableResult
+    public func registerForBackgroundUpdates() -> Bool {
+        // register the bg task
+        let result = BGTaskScheduler.shared.register(forTaskWithIdentifier: bgTaskIdentifier, using: nil) { task in
+            self.handleBackgroundRefresh(task: task as! BGAppRefreshTask)
+        }
+        
+        // schedule a notification when application enters background
+        if bgObserver == nil {
+            bgObserver = NotificationCenter.default.addObserver(forName: UIApplication.willResignActiveNotification, object: nil, queue: .main) { [weak self] _ in
+                self?.scheduleNextBackgroundRefresh()
+            }
+        }
+        
+        return result
+    }
+    
+    @available(iOS 13.0, *)
+    private func handleBackgroundRefresh(task: BGAppRefreshTask) {
```

Comment:
If you change the type of `task` to `BGTask`, you wouldn't need to cast on line `232`.

```suggestion
    private func handleBackgroundRefresh(task: BGTask) {
```

Reply:
The way you are recommending this is the way I actually wrote it as well!  The problem is I wasn't seeing the background refresh work and every sample I looked at had it written this way and I started wondering if there was something in apples code checking for the parameter type. I saw at least 3 examples of the code written like this - including apple's own sample code, so I changed it to this.

Reply:
Weird! I can't think of why that would make a difference, but apparently it does. I just saw that it compiled; I didn't actually test it. Thank you for the explanation.

Reply:
I don't know if it did actually make a difference in the end. The way to test this is not great and I didn't realize it didn't work on the simulator at all until later.